test/integration/internal-extension.js

```diff
@@ -81,16 +82,27 @@ test('internal extension', t => {
     };
     t.deepEqual(goBlockInfo, expectedBlockInfo);
 
-    // There should be 2 menus - one is an array, one is the function to call.
-    t.equal(vm.runtime._blockInfo[0].menus.length, 2);
-    // First menu has 3 items.
-    t.equal(
-        vm.runtime._blockInfo[0].menus[0].json.args0[0].options.length, 3);
-    // Second menu is a dynamic menu and therefore should be a function.
-    t.type(
-        vm.runtime._blockInfo[0].menus[1].json.args0[0].options, 'function');
-
-    t.end();
+    // There should be at least one extension loaded, but there could be more
+    // loaded if we have core extensions defined in src/virtual-machine.
+    t.ok(vm.runtime._blockInfo.length > 0);
+
+    for (const extensionInfo of vm.runtime._blockInfo) {
+        if (extensionInfo.id !== 'testInternalExtension') continue;
```

Comment:
👍
You could slightly simplify this using something like `const extensionInfo = vm.runtime._blockInfo.find(x => x.id === 'testInternalExtension');` but the for/continue setup is already much better than the sometimes-broken version we had before :)

Reply:
That's a great suggestion! Done.